### ISSUE TITLE
Instructions for setting up imagenet dataset on AWS storage

### ISSUE THREAD

**Issue by kbhardwa** [diux-dev/imagenet18#17]
Hi,

I am trying to use train.py for one p3.16xlarge instance and I am getting Access Denied Errors (likely because I do not have access to EFS):

```
Creating security group ncluster
Creating keypair ncluster-ubuntu
efs_client.describe_file_systems failed with <class 'botocore.exceptions.ClientError'>(An error occurred (AccessDeniedException) when calling the DescribeFileSystems operation: User: arn:aws:iam::xxxxxxxxxxxx:user/xxxxxxxx is not authorized to perform: elasticfilesystem:DescribeFileSystems on the specified resource), retrying
```
Is this error occurring because I do not have access to AWS? Can you please provide instructions in README on what specifically to setup for storage? Do we setup an EFS and download imagenet in it at the following path: `~/data/imagenet`? Or will the code do all that on its own?

Also, I see that we need access to io2 disk on AWS as given here: https://github.com/diux-dev/imagenet18/blob/59a8f25171fb8cede51db9187a32fc8f802384a0/train.py#L161.

However, I could not find an io2 storage on AWS (https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSVolumeTypes.html) and could find only io1. The aws_backend.py code (in ncluster) also talks about setting up io1 (as given here: https://github.com/diux-dev/ncluster/blob/839514a1536b6157d59d043a1ff4dac45c2a4ffd/ncluster/aws_backend.py#L770-L776). Am I missing something? Can you please update instructions for what all we need to do to process the imagenet dataset for train.py?

Thanks a lot.
PS: initially, we setup an general purpose EBS, downloaded imagenet on it, and mounted it on this p3.16xlarge instance. Is that not needed?

**Comment by yaroslavvb** [comment]
You do not need to setup any storage, however, you need to have permission to create EFS resources on AWS. Ask your admin to add this permission for you.

This code assumes admin-level permissiojns

**Comment by kbhardwa** [comment]
Awesome! So do I need just the EFS storage? Is there a separate permission for io1/io2? And just to confirm, imagenet downloading, preprocessing etc. will happen all on its own?

**Comment by yaroslavvb** [comment]
I've updated https://github.com/diux-dev/imagenet18/blob/master/README.md
You don't need EFS storage, but you do need permissions. The code will create EFS storage and do everything else

**Comment by kbhardwa** [comment]
Thanks a lot @yaroslavvb . This is indeed great! Will save us a lot of time! (and yes, I updated my comment for permissions; sorry my bad!)

**Comment by yaroslavvb** [comment]
I don't believe there's separate permission for io2, but there's a limit of how many IOPS you can request. You can see the limit in the "Limits" section of AWS console

**Comment by kbhardwa** [comment]
Sounds good!
